rheem-platforms/rheem-giraph/src/main/java/org/qcri/rheem/giraph/operators/GiraphPageRankOperator.java

```diff
@@ -0,0 +1,258 @@
+                "rheem.giraph.pagerank.load.main", configuration
+        ));
+        mainExecutionLineage.addPredecessor(inputFileChannelInstance.getLineage());
+
+        final ExecutionLineageNode outputExecutionLineage = new ExecutionLineageNode(operatorContext);
+        outputExecutionLineage.add(LoadProfileEstimators.createFromSpecification(
+                "rheem.giraph.pagerank.load.output", configuration
+        ));
+        outputChannelInstance.getLineage().addPredecessor(outputExecutionLineage);
+
+        return mainExecutionLineage.collectAndMark();
+    }
+
+
+    @Override
+    public Platform getPlatform() {
+        return GiraphPlatform.getInstance();
+    }
+
+    @Override
+    public Collection<String> getLoadProfileEstimatorConfigurationKeys() {
+        return Arrays.asList("rheem.giraph.pagerank.load.main", "rheem.giraph.pagerank.load.output");
+    }
+
+    @Override
+    public List<ChannelDescriptor> getSupportedInputChannels(int index) {
+        return Collections.singletonList(FileChannel.HDFS_TSV_DESCRIPTOR);
+    }
+
+    @Override
+    public List<ChannelDescriptor> getSupportedOutputChannels(int index) {
+        return Collections.singletonList(StreamChannel.DESCRIPTOR);
+    }
+
+
+    public void setPathOut(String path, Configuration configuration){
+        if(path == null && configuration != null) {
+            path = configuration.getStringProperty("rheem.giraph.hdfs.tempdir");
+        }
+        this.path_out = path;
+    }
+
+    public String getPathOut(Configuration configuration){
+        if(this.path_out == null){
+            setPathOut(null, configuration);
+        }
+        return this.path_out;
+    }
+
+
+    private Stream<Tuple2<Long, Float>> createStream(String path) {
+        return this.streamLines(path).map(line -> {
+            String[] part = line.split("\t");
+            return new Tuple2<>(Long.parseLong(part[0]), Float.parseFloat(part[1]));
+        });
+    }
+
+
+    /**
+     * Creates a {@link Stream} of a lines of the file.
+     *
+     * @param path of the file
+     * @return the {@link Stream}
+     */
+    private Stream<String> streamLines(String path) {
+        final FileSystem fileSystem = FileSystems.getFileSystem(path).orElseThrow(
+                () -> new IllegalStateException(String.format("No file system found for %s", path))
+        );
+        try {
+            Iterator<String> lineIterator = this.createLineIterator(fileSystem, path);
+            return StreamSupport.stream(Spliterators.spliteratorUnknownSize(lineIterator, 0), false);
+        } catch (IOException e) {
+            throw new RheemException(String.format("%s failed to read %s.", this, path), e);
+        }
+
+    }
+
+    /**
+     * Creates an {@link Iterator} over the lines of a given {@code path} (that resides in the given {@code fileSystem}).
+     */
+    private Iterator<String> createLineIterator(FileSystem fileSystem, String path) throws IOException {
```

Comment:
This seems to be copied from `JavaTsvFileSource.createLineIterator`. It is probably better to move this function to some `FileUtils` class or so, e.g., in the `rheem-core/src/main/java/org/qcri/rheem/core/util/fs` folder.